File: app/src/main/java/com/sample/chen/multiplestatusview/MainActivity.java
package com.sample.chen.multiplestatusview;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import com.github.clans.fab.FloatingActionMenu;
import com.sample.chen.library.MultipleStatusView;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private MultipleStatusView mMultipleStatusView;
    private FloatingActionMenu mFloatingActionMenu;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        mMultipleStatusView = (MultipleStatusView) findViewById(R.id.multiple_status_view);
        mFloatingActionMenu = (FloatingActionMenu) findViewById(R.id.main_fab_menu);
        com.github.clans.fab.FloatingActionButton mLoadingFab = (com.github.clans.fab.FloatingActionButton) findViewById(R.id.main_fab_loading);
        com.github.clans.fab.FloatingActionButton mEmptyFab = (com.github.clans.fab.FloatingActionButton) findViewById(R.id.main_fab_empty);
        com.github.clans.fab.FloatingActionButton mErrorFab = (com.github.clans.fab.FloatingActionButton) findViewById(R.id.main_fab_error);
        com.github.clans.fab.FloatingActionButton mNoNetworkFab = (com.github.clans.fab.FloatingActionButton) findViewById(R.id.main_fab_no_network);
        com.github.clans.fab.FloatingActionButton mContentFab = (com.github.clans.fab.FloatingActionButton) findViewById(R.id.main_fab_content);
        mLoadingFab.setOnClickListener(this);
        mEmptyFab.setOnClickListener(this);
        mErrorFab.setOnClickListener(this);
        mNoNetworkFab.setOnClickListener(this);
        mContentFab.setOnClickListener(this);

        mMultipleStatusView.setOnActionListener(new MultipleStatusView.OnActionListener() {
            @Override
            public void onAction(View view) {
                Toast.makeText(getBaseContext(), "OnActionListener", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.main_fab_loading:
                mMultipleStatusView.loading();
                break;
            case R.id.main_fab_empty:
                mMultipleStatusView.empty();
                break;
            case R.id.main_fab_error:
                mMultipleStatusView.custom(getResources().getDrawable(R.mipmap.ic_error),
                        "error","retry");
                break;
            case R.id.main_fab_no_network:
                mMultipleStatusView.custom(getResources().getDrawable(R.mipmap.ic_no_network),
                        "no network","retry");
                break;
            case R.id.main_fab_content:
                mMultipleStatusView.dismiss();
                break;
        }
        mFloatingActionMenu.toggle(false);
    }
}

File: app/src/main/java/com/sample/chen/multiplestatusview/MainActivity.kt
package com.sample.chen.multiplestatusview

import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.View
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast

import com.github.clans.fab.FloatingActionMenu
import com.sample.chen.library.MultipleStatusView

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private var mMultipleStatusView: MultipleStatusView? = null
    private var mFloatingActionMenu: FloatingActionMenu? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById(R.id.toolbar) as Toolbar?
        setSupportActionBar(toolbar)

        mMultipleStatusView = findViewById(R.id.multiple_status_view) as MultipleStatusView?
        mFloatingActionMenu = findViewById(R.id.main_fab_menu) as FloatingActionMenu?
        val mLoadingFab = findViewById(R.id.main_fab_loading) as com.github.clans.fab.FloatingActionButton?
        val mEmptyFab = findViewById(R.id.main_fab_empty) as com.github.clans.fab.FloatingActionButton?
        val mErrorFab = findViewById(R.id.main_fab_error) as com.github.clans.fab.FloatingActionButton?
        val mNoNetworkFab = findViewById(R.id.main_fab_no_network) as com.github.clans.fab.FloatingActionButton?
        val mContentFab = findViewById(R.id.main_fab_content) as com.github.clans.fab.FloatingActionButton?
        mLoadingFab!!.setOnClickListener(this)
        mEmptyFab!!.setOnClickListener(this)
        mErrorFab!!.setOnClickListener(this)
        mNoNetworkFab!!.setOnClickListener(this)
        mContentFab!!.setOnClickListener(this)

        mMultipleStatusView!!.setOnActionListener(object : MultipleStatusView.OnActionListener {
            override fun onAction(view: View) {
                Toast.makeText(baseContext, "OnActionListener", Toast.LENGTH_SHORT).show()
            }
        })
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId


        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)

    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.main_fab_loading -> mMultipleStatusView!!.loading()
            R.id.main_fab_empty -> mMultipleStatusView!!.empty()
            R.id.main_fab_error -> mMultipleStatusView!!.custom(resources.getDrawable(R.mipmap.ic_error),
                    "error", "retry")
            R.id.main_fab_no_network -> mMultipleStatusView!!.custom(resources.getDrawable(R.mipmap.ic_no_network),
                    "no network", "retry")
            R.id.main_fab_content -> mMultipleStatusView!!.dismiss()
        }
        mFloatingActionMenu!!.toggle(false)
    }
}
